### PR TITLE
[FRS-15] Add more metrics

### PR DESCRIPTION
**Pull request by wsry**
## What is the purpose of the change

This solves #15 . This patch adds more metrics for better cluster monitoring. Please refer to the document for more information: https://github.com/flink-extended/flink-remote-shuffle/blob/main/docs/user_guide.md#operations.

## Brief change log

  - This patch adds more metrics for better cluster monitoring.

## Verifying this change

This change added tests.

### CODE REVIEW

File: shuffle-transfer/src/main/java/com/alibaba/flink/shuffle/transfer/ReadServerHandler.java [hunk 1]
```diff
@@ -107,13 +107,13 @@ protected void channelRead0(ChannelHandlerContext ctx, TransferMessage msg) {
     @Override
     public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
         super.channelRegistered(ctx);
-        NetworkMetrics.numReadingConnections().inc();
+        NetworkMetricsUtil.registerNumReadingConnections().inc();
```
Comment (TanYuxin-tyx) [inline shuffle-transfer/src/main/java/com/alibaba/flink/shuffle/transfer/ReadServerHandler.java:110]: Why do we rename these metrics methods?
For example, here is counting for reading connections, is it better to use `numReadingConnections`?

**Review by TanYuxin-tyx — COMMENTED** [review]
@wsry Thanks for adding more metrics. I only left a minor comment about the PR.

File: shuffle-transfer/src/main/java/com/alibaba/flink/shuffle/transfer/ReadServerHandler.java (hunk 1 above)
Comment (wsry) [reply shuffle-transfer/src/main/java/com/alibaba/flink/shuffle/transfer/ReadServerHandler.java:110]: The rename is for consistency and easier understanding. We have two phases here, first register a metric, second update the metric. The register rename change is for easier understanding of first step. I agree that it is a little misleading here. I will split the two phases here into two line and make the Counter a global variable which will make it easier to understand.

**Review by TanYuxin-tyx — APPROVED** [review]
After fixing the minor comments, the change looks good to me. @wsry Thanks for adding these useful metrics.

**Pull request merged**